presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -364,14 +364,10 @@ public TableLayoutResult getLayout(Session session, TableHandle table, Constrain
         CatalogMetadata catalogMetadata = getCatalogMetadata(session, connectorId);
         ConnectorMetadata metadata = catalogMetadata.getMetadataFor(connectorId);
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
-        List<ConnectorTableLayoutResult> layouts = metadata.getTableLayouts(connectorSession, connectorTable, constraint, desiredColumns);
+        ConnectorTableLayoutResult layout = metadata.getTableLayoutForConstraint(connectorSession, connectorTable, constraint, desiredColumns);
```

Comment:
I would try to ensure that the results between `getTableLayouts` and `getTableLayoutForConstaint` are consistent, otherwise it might be confusing to connector authors.

Remember that this method is also used by connectors not even known to Presto (i.e. external libraries).  So we need to accomodate people who are slow to adopt to the new API.